frontend/src/components/user/trombi/Trombi.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from "react";
+import { Pagination } from "../../utils/Pagination";
+import { api, useBetterQuery } from "../../../services/apiService";
+import { UserAvatarCard } from "../../utils/avatar/UserAvatarCard";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+import Container from "react-bootstrap/Container";
+import { PageTitle } from "../../utils/PageTitle";
+import { Link } from "react-router-dom";
+import Select, { OptionsType, ValueType } from "react-select";
+import InputGroup from "react-bootstrap/InputGroup";
+import { DebounceInput } from "react-debounce-input";
+
+export const Trombi = () => {
+    const { data: promotions } = useBetterQuery<{
+        promotions: string[];
+    }>(["listPromotions"], api.users.listPromotions, {
+        refetchOnWindowFocus: false,
+    });
+
+    // react-select requires an `options` field of a special type. These hooks handle this.
+    const [promotionsOptions, setPromotionsOptions] = useState<
+        OptionsType<{ value: string; label: string }>
+    >([]);
+
+    useEffect(() => {
+        if (promotions) {
+            setPromotionsOptions(
+                promotions.promotions.map((promotion) => {
+                    return { value: promotion, label: promotion };
+                })
+            );
+        }
+    }, [promotions]);
+
+    // The search key sent to `api.users.list`. It is debounced thanks to the `searchInputValue` hooks.
+    const [searchKey, setSearchKey] = useState<string>("");
```

Comment:
TS compiler is able to infer the type from default value
```suggestion
    const [searchKey, setSearchKey] = useState("");
```